src/script/script.cpp

```diff
@@ -262,6 +263,21 @@ bool CScript::IsPayToScriptHash() const
     return p2sh || p2shWithReplay;
 }
 
+bool CScript::IsPayToPublicKey() const
+{
+    // Extra-fast test for pay-to-pubkey CScripts:
+
+    // Check if this script is P2PK without OP_CHECKBLOCKATHEIGHT
+    bool p2pk = (this->size() == 35 &&
+               (*this)[0] == 0x21 &&
+               (*this)[34] == OP_CHECKSIG);
+
+     // Check if this script is P2PKH with OP_CHECKBLOCKATHEIGHT
+    bool p2pkWithReplay = false; //TODO: check format
```

Comment:
Can we solve this TODO ?

Reply:
I have added following code, trying to replicate from p2pkh->p2pkhWithReplay and p2sh->p2shWithReplay:


`

     // Check if this script is P2PKH with OP_CHECKBLOCKATHEIGHT

     // The overall size should not be more then 74:

     // 35 bytes for common P2PK + 1 byte size + 32 bytes of block hash + 1 byte size + 4 bytes of block height + 1 byte opcode

    bool p2pkWithReplay = (size() > 35 &&
                           size() < 75 &&
                           at(0) == 0x21 &&
                           at(34) == OP_CHECKSIG &&
                           at(size() - 1) == OP_CHECKBLOCKATHEIGHT);
`

unfortunately I was not able to find any useful documentation supporting this script scheme, @ptagl can you please confirm on correctness?

Reply:
ping @ptagl

Reply:
It seems fine to me as it reflects the same logic in the `Solver` function inside `standard.cpp`.

Reply:
Agree